test/cqlpy/test_lwt.py

```diff
@@ -103,6 +108,17 @@ def test_lwt_with_batch_conflict_2(cql, table1):
     for r in rs:
         assert r.applied == False
 
+# Moreover, there are cases where Cassandra prevents mixing
+# IF EXISTS with different conditions such as IF r=1, despite
+# the fact that the conditions are not contradictory.
+def test_lwt_with_batch_conflict_3(cql, table1, scylla_only):
+    p = unique_key_int()
+    cql.execute(f'INSERT INTO {table1}(p,c,r) VALUES ({p},1,1)')
+    rs = list(cql.execute(f'BEGIN BATCH UPDATE {table1} SET r=10 WHERE p={p} AND c=1 IF r=1; UPDATE {table1} SET r=20 WHERE p={p} AND c=1 IF EXISTS;  APPLY BATCH;'))
+    # Cassandra fails with: "Cannot mix IF conditions and IF EXISTS for the same row"
+    for r in rs:
+        assert r.applied == True
```

Comment:
Nice catch. It's weird that Cassandra won't apply this **successful** batch. I wonder if it's not even a bug that should be reported to them. But never mind :-)